tests/integ/test_testing.py

```diff
@@ -0,0 +1,259 @@
+        ]
+    )
+    df4 = session.create_dataframe([], schema=schema4)
+    assert_dataframe_equal(df1, df4)
+    schema5 = StructType(
+        [
+            StructField("id", IntegerType()),
+            StructField('"NAME"', StringType()),
+            StructField("value", FloatType()),
+        ]
+    )
+    df5 = session.create_dataframe([], schema=schema5)
+    assert_dataframe_equal(df1, df5)
+
+
+def test_schema_datatype(session):
+    schema1 = StructType(
+        [
+            StructField("id", IntegerType()),
+            StructField("name", StringType()),
+            StructField("value", FloatType()),
+        ]
+    )
+    df1 = session.create_dataframe([], schema=schema1)
+    schema2 = StructType(
+        [
+            StructField("id", FloatType()),
+            StructField("name", StringType()),
+            StructField("value", FloatType()),
+        ]
+    )
+    df2 = session.create_dataframe([], schema=schema2)
+    with pytest.raises(AssertionError, match="Different schema:"):
+        assert_dataframe_equal(df1, df2)
+    schema3 = StructType(
+        [
+            StructField("id", LongType()),
+            StructField("name", StringType()),
+            StructField("value", DoubleType()),
+        ]
+    )
+    df3 = session.create_dataframe([], schema=schema3)
+    assert_dataframe_equal(df1, df3)
+    schema4 = StructType(
+        [
+            StructField("id", IntegerType()),
+            StructField("name", StringType(2)),
+            StructField("value", FloatType()),
+        ]
+    )
+    df4 = session.create_dataframe([], schema=schema4)
+    with pytest.raises(AssertionError, match="Different schema:"):
+        assert_dataframe_equal(df1, df4)
+
+
+def test_schema_nullable(session):
+    schema1 = StructType(
+        [
+            StructField("id", IntegerType()),
+            StructField("name", StringType()),
+            StructField("value", FloatType()),
+        ]
+    )
+    df1 = session.create_dataframe([], schema=schema1)
+    schema2 = StructType(
+        [
+            StructField("id", IntegerType(), nullable=False),
+            StructField("name", StringType()),
+            StructField("value", FloatType()),
+        ]
+    )
+    df2 = session.create_dataframe([], schema=schema2)
+    with pytest.raises(AssertionError, match="Different schema:"):
+        assert_dataframe_equal(df1, df2)
+
+
+def test_input_not_snowpark_df(session):
+    try:
+        import pandas
+    except ImportError:
+        pytest.skip("No pandas is available")
+
+    data = [1]
+    df = session.create_dataframe(data)
+    pandas_df = pandas.DataFrame(data)
+    with pytest.raises(TypeError, match="must be a Snowpark DataFrame"):
+        assert_dataframe_equal(df, data)
+    with pytest.raises(TypeError, match="must be a Snowpark DataFrame"):
+        assert_dataframe_equal(pandas_df, df)
+
+
+def test_assertDataFrameEqual():
+    assert assert_dataframe_equal == assertDataFrameEqual
```

Comment:
I don't understand this test

Reply:
I'm following this https://github.com/snowflakedb/snowpark-python/blob/91a86f404c9bf6430e9ae32d7399bbd8505f35fe/tests/unit/test_dataframe.py#L51